jfr-streaming/src/test/java/io/opentelemetry/contrib/jfr/metrics/MemoryUsageMetricTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jfr.metrics;
+
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_COMPRESSED_CLASS_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_G1_EDEN_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_G1_SURVIVOR_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_JAVA_HEAP_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_METASPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_PS_EDEN_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_PS_OLD_GEN;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.ATTR_PS_SURVIVOR_SPACE;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.BYTES;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_DESCRIPTION_MEMORY;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_DESCRIPTION_MEMORY_AFTER;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_NAME_MEMORY;
+import static io.opentelemetry.contrib.jfr.metrics.internal.Constants.METRIC_NAME_MEMORY_AFTER;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.SumData;
+import org.assertj.core.api.ThrowingConsumer;
+import org.junit.jupiter.api.Test;
+
+class MemoryUsageMetricTest extends AbstractMetricsTest {
+
+  private void check(ThrowingConsumer<MetricData> attributeCheck) {
+    waitAndAssertMetrics(
+        metric ->
+            metric
+                .hasName(METRIC_NAME_MEMORY)
+                .hasUnit(BYTES)
+                .hasDescription(METRIC_DESCRIPTION_MEMORY)
+                .satisfies(attributeCheck),
+        metric ->
+            metric
+                .hasName(METRIC_NAME_MEMORY_AFTER)
+                .hasUnit(BYTES)
+                .hasDescription(METRIC_DESCRIPTION_MEMORY_AFTER)
+                .satisfies(attributeCheck));
+  }
+
+  /**
+   * This is a basic test for process.runtime.jvm.memory.usage and
+   * process.runtime.jvm.memory.usage_after_last_gc metrics.
+   */
+  @Test
+  void shouldHaveMemoryUsageMetrics() {
+    System.gc();
+    if (garbageCollectors.contains("G1 Young Generation")) {
```

Comment:
Shouldn't the garbage collector in use for a given build be deterministic? 

Consider breaking out this test by garbage collector and using [assumeThat](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractGrpcTelemetryExporterTest.java#L241-L250) to only verify the conditions if the assumption is true.

Reply:
I'm a bit confused. Isn't the GC specified at runtime instead of buildtime? I'm looking for a way to specify a different GC for each test so I can ensure that metrics for the various pools are correct. Right now, I haven't figured out how to do that.  Additionally, I'm not sure how to specify a particular GC without passing it as a runtime option (so the JVM chooses a default depending on the system). This is why the tests are set up like this. 

I can break out these tests by GC, but they still won't be that useful until the points above are resolved.

Reply:
You can create different test suites that specify different JVM args. Check out this [commit](https://github.com/jack-berg/opentelemetry-java-contrib/commit/0c83192f98ef17ffe4a1aeae3d1b191241e11dab) where I introduce a new test suite called "testShenandoah", which runs with JVM args `-XX:+UnlockExperimentalVMOptions XX:+UseShenandoahGC`. Could use the same technique to create a different test suite for each GC that needs to be tested.

Reply:
That's exactly what I was looking for!